homeassistant/components/wled/select.py

```diff
@@ -0,0 +1,100 @@
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import WLEDDataUpdateCoordinator
+from .helpers import wled_exception_handler
+from .models import WLEDEntity
+
+PARALLEL_UPDATES = 1
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up WLED select based on a config entry."""
+    coordinator: WLEDDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    update_segments = partial(
+        async_update_segments,
+        coordinator,
+        {},
+        async_add_entities,
+    )
+    coordinator.async_add_listener(update_segments)
+    update_segments()
+
+
+class WLEDPaletteSelect(WLEDEntity, SelectEntity):
+    """Defines a WLED Palette select."""
+
+    _attr_icon = "mdi:palette-outline"
+    _segment: int
+    _attr_entity_registry_enabled_default = False
+
+    def __init__(self, coordinator: WLEDDataUpdateCoordinator, segment: int) -> None:
+        """Initialize WLED ."""
+        super().__init__(coordinator=coordinator)
+
+        # Segment 0 uses a simpler name, which is more natural for when using
+        # a single segment / using WLED with one big LED strip.
+        self._attr_name = (
+            f"{coordinator.data.info.name} Segment {segment} Color Palette"
+        )
+        if segment == 0:
+            self._attr_name = f"{coordinator.data.info.name} Color Palette"
+
+        self._attr_unique_id = f"{coordinator.data.info.mac_address}_palette_{segment}"
+        self._attr_options = [
+            palette.name for palette in self.coordinator.data.palettes
+        ]
+        self._segment = segment
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        try:
+            self.coordinator.data.state.segments[self._segment]
+        except IndexError:
+            return False
+
+        return super().available
+
+    @property
+    def current_option(self) -> str | None:
+        """Return the current selected color palette."""
+        return self.coordinator.data.state.segments[self._segment].palette.name
+
+    @wled_exception_handler
+    async def async_select_option(self, option: str) -> None:
+        """Set WLED segment to the selected color palette."""
+        await self.coordinator.wled.segment(segment_id=self._segment, palette=option)
+
+
+@callback
+def async_update_segments(
+    coordinator: WLEDDataUpdateCoordinator,
+    current: dict[int, WLEDPaletteSelect],
+    async_add_entities,
+) -> None:
+    """Update segments."""
+    segment_ids = {segment.segment_id for segment in coordinator.data.state.segments}
+    current_ids = set(current)
+
+    new_entities = []
+
+    # Process new segments, add them to Home Assistant
+    for segment_id in segment_ids - current_ids:
+        current[segment_id] = WLEDPaletteSelect(coordinator, segment_id)
```

Comment:
There's no reason to store the entity. It's enough to store the segment id in a set.

Reply:
yeah, that is a nice more generic improvement, thanks! 👍